### DISCUSSION TITLE
Cannot get capacitor plugins to be recognized as implimented

### DISCUSSION THREAD

**Discussion by mauzilla** [quasarframework/quasar#17500] [CLI - Capacitor / Cordova modes] [answered]
Using quasar with capacitor 6. I want to use the Preferences plugin. I have:

-  in the src-capacitor/ directory installed @capacitor/Preferences (no issues)
- When I run quasar dev -m capacitor -T Android I can see the alias is created
- I have also tried npx cap sync from within src-capacitor, and it recognizes the plugin (plugin is also installed in node-modules/@capacitor/ in src-capacitor/

When I try and use the plugin, I get:

http://10.0.10.133:9500/src-capacitor/node_modules/@capacitor/core/dist/index.js - Line 188 - Msg: Uncaught (in promise) Error: "Preferences" plugin is not implemented on android

I have within Android Studio cleared caches, clean and rebuild, even restarted my machine. Gradle sync, you name it, it simply does not accept the plugin. Is anyone able to help me out, I have been battling now for hours. Even tried to delete the node-modules directory in src-capacitor and did a fresh npm install, no difference.

Capacitor is cleary installed as I am able to interact with the app in Android Studio and the remotedevtools in Chrome, but for whatever reason the Capacitor plugins does seem to be loaded.

**Comment by yusufkandemir** [comment, marked as answer]
From [my answer on Discord](https://discord.com/channels/415874313728688138/856517527700766751/1282726034789105726):

> It works fine for me, here's the steps I took:
> ```sh
> $ npm init quasar
> # created an app-vite v2 JS app
> $ npx quasar mode add capacitor
> $ cd src-capacitor
> $ npm add @capacitor/preferences
> # also add some code in IndexPage that imports and uses the plugin
> $ cd ..
> $ npx quasar dev -m capacitor -T android
> # generates the web assets
> # updates the relevant files
> # runs cap sync android
> ```

The OP confirmed a clean project works. Afterward, they said it worked on their project after restarting things.
